File: demo-app/src/types/appUserRegistration.ts
// App User Registration Types
export interface AppUserRegistration {
  id: string;
  appId: string;
  userId: string;
  provider: string;
  externalId: string;
  status: number;
  createdDate: string;
  modifiedDate: string;
  autoID: number;
}

export interface AppUserRegistrationAdd {
  appId: string;
  userId: string;
  provider: string;
  externalId: string;
}

File: SaasV2.Entity/AppUserRegistration.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SaasV2.Entity
{
    // Kullanıcının belirli bir App'e kaydı
    public class AppUserRegistration : BaseEntity
    {
        public Guid AppId { get; set; }
        public Guid UserId { get; set; }          // AppUser.Id
        public DateTime RegisteredAt { get; set; }
        public string Provider { get; set; }      // e-posta, apple, google vs. (opsiyonel)
        public string ExternalId { get; set; }    // dış sağlayıcı kimliği (opsiyonel)
    }
}
